RevenueCatUI/Views/PurchaseButton.swift

```diff
@@ -59,39 +59,29 @@ struct PurchaseButton: View {
         self.purchaseHandler = purchaseHandler
     }
 
-    @Environment(\.userInterfaceIdiom)
-    var userInterfaceIdiom
-
     var body: some View {
         self.button
     }
 
     private var button: some View {
         AsyncButton {
             guard !self.purchaseHandler.actionInProgress else { return }
+            guard !self.package.currentlySubscribed else { return }
```

Comment:
Doing this instead of marking the button as `disabled`, which would display a spinner now.